### ISSUE TITLE
Simplify issue reporting

### ISSUE THREAD

**Issue by ctoth** [nvaccess/nvda#13703]
### Describe the solution you'd like

In the help menu, I would like to report an issue to NVDA. It would include all of the required information in a bug/issue report without me doing anything special using the GitHub API and appropriate system-info-gathering primitives. 

### Describe alternatives you've considered
Copy/pasting is silly

### Additional context
Many bug report forms for NVDA include information which may be automatically derived from within the NVDA program such as the operating system and NVDA versions. There should be a way to report a bug from within NVDA that places the bug report on GitHub but also automatically collects this information.

### UI
- In the help menu, place a Report an Issue option
- Add a checkbox to automatically include logs
- Radio buttons for GitHub issue types

**Comment by Brian1Gaff** [comment]
I seem to recall this debate before, but the main problem was personal data 
in the log which the user does not actually know is there before the report 
goes out.
 Brian

***@***.***
Sent via blueyonder.
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message ----- 
From: "Christopher Toth" ***@***.***>
To: "nvaccess/nvda" ***@***.***>
Cc: "Subscribed" ***@***.***>
Sent: Tuesday, May 17, 2022 8:04 AM
Subject: [nvaccess/nvda] Simplify issue reporting (Issue #13703)

**Comment by ctoth** [comment]
This is why the user must explicitly check a box to include the log. This would be no different than uploading a log upon request.

**Comment by XLTechie** [comment]
This has been considered by various people before. I agree that it has some obvious advantages from a user prospective, and from the developer prospective with the ability to have more granular control of what's in the issue.
When I looked into doing it, some of the obstacles were:

* In order to post an issue to GitHub, the user must first have a GitHub account. AFAIK, that can only be done via the GH website.
* In order to post an issue via API, I believe the user must obtain an API key, and authenticate with it. There's probably an Oauth2 process in place for that to authenticate a particular instance of NVDA, but it's a lot to go through for reporting a bug.
* Even if we implemented an alternative process, such as having NV Access host a service on one of their servers, that accepted incoming unauthenticated bug reports, and reported them with the @nvaccessAuto  account, those reports would then not be connected to the original user. That would also be a headache for maintenance, security, spam prevention, etc.
* Doing any of this would also lead to users not being positioned to first search for existing bug reports. I would anticipate that the number of duplicates needing to be flagged and closed would rise.

Do you have a proposal for overcoming these difficulties?

**Comment by christopherpross** [comment]
Well I would prefer to have such a option my thoughts are:
1. Yeah, we need a account for posting issues, we have a nvda-access account or nvda-issue-bot or something else. A little python/go server could accept the issue report and could post that on github.
2. personal informations in the logs, we could store the logs on the server and provide a link to it, this link can only open by devs. Or we say that the logs can contain sensitive information and let the user choose, whether to upload one or maybe which log, nvda could archivate more than one... the last option for that is that the little server sends the log files to a only-dev mail adress with the issue number in the subject...
3. The problem with the relation to the orginal poster is a problem that is right. Maybe we could create a user id and identify users with that?
4. The searching of same issues is a problem too. My solution would: We let users create issues, but we need people which look in the issues, maybe this would not published until a user looked over it and aprove it as real issue? 

In fact this way for reporting bugs contains many obstacles and problems. It is not as easy at seems.

**Comment by Brian1Gaff** [comment]
So, at the moment then, when we fill in the template, which is not as hard 
as you might think it is, how does it get to the correct place and get 
checked for duplicates?
 When I do a template now, I tend to open the  ticket, put in the various 
info, then cut and paste the template to an edditor and fill it out there, 
then  just delete the whole template on line and paste the filed in one over 
the top. Maybe an automated not on line system could be the best way forward 
and the ticket number would get allocated when you did the upload instead.
 Brian

***@***.***
Sent via blueyonder.
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message ----- 
From: "Christopher Proß" ***@***.***>
To: "nvaccess/nvda" ***@***.***>
Cc: "Brian Gaff" ***@***.***>; "Comment" 
***@***.***>
Sent: Wednesday, May 18, 2022 10:00 AM
Subject: Re: [nvaccess/nvda] Simplify issue reporting (Issue #13703)

**Comment by seanbudd** [comment]
Duplicate of issue https://github.com/nvaccess/nvda/issues/4033, https://github.com/nvaccess/nvda/issues/7745 and https://github.com/nvaccess/nvda/issues/8685. Keeping this one in favour of the discussion in these as there is a clearer summary of the problem and issues with the proposed solution.

**Comment by seanbudd** [comment]
GitHub issues are not a user support channel - we encourage users to contact [support](https://www.nvaccess.org/product/nvda-telephone-support/).
This change would lead to bug reports being user centric, rather than developer centric.
If GitHub was flooded with these issues NV Access would not have the capacity to triage them.

**Comment by seanbudd** [comment]
As the proposal stands, with the issues highlighted in my earlier comment and in https://github.com/nvaccess/nvda/issues/13703#issuecomment-1129256020, we don't forsee this being accepted in NVDA.
